ext/dom/tests/registerNodeClass_abstract_class.phpt

```diff
@@ -0,0 +1,24 @@
+--TEST--
+registerNodeClass() with an abstract class should fail
+--EXTENSIONS--
+dom
+--FILE--
+<?php
+
+abstract class Test extends DOMElement {
+    abstract function foo();
+}
+
+$dom = new DOMDocument;
+
+try {
+    $dom->registerNodeClass("DOMElement", "Test");
```

Comment:
Should there be another test for passing a enum?

Reply:
An enum cannot pass the `instanceof_function(ce, basece)` check. i.e. you'll get `Argument #2 ($extendedClass) must be a class name derived from DOMElement or null, **enum name here** given`.